### PR TITLE
fix: Support Dark theme based on the web browser's theme

### LINKED ISSUE #16: Support Dark theme based on the web browser's theme
Now the project doesn't switch to dark theme when the browser's theme is set to dark, we have to make it compatible for dark theme. 
To better test this, use edge browser and switch the theme between light and dark.

### PR DESCRIPTION
**Pull request by SankalpHaritash21**
## Summary

Adding Dark Theme feature in add blog, home page. 

## Description

In this project adding dark theme feature according to the browser theme.

## Images

Add blog page
add

Home page

home

Details page

detail

## Issue(s) Addressed

- Closes #16 

## Prerequisites

- [X] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

**Comment by SankalpHaritash21** [comment]
Sir number of div.
And Sorry for the delay.
If any change needed plz let me know I will make that changes.
Sir In some places I used little bit differet

File: frontend/tailwind.config.js
Comment (chinmaykunkikar) [inline frontend/tailwind.config.js:10]: While everything else looks good to me. The name of this variable is the only thing that bothers me.. a lot 😛 

You can probably translate `background_card_textfield` to something like this:

```js
extend: {
      colors: {
        dark: {
          background: {
		DEFAULT: 'var(--color-dark-background)',
		card: {
			textfield: 'var(--color-dark-textfield-card)'
		}
	},
```

This will allow you to use `dark:bg-dark-background-card-textfield`

But, if feel "background" here is redundant and I think is against Tailwind's naming conventions.
How about if we do `dark:bg-dark-textfield`? Like this:

```js
extend: {
      colors: {
        dark: {
		DEFAULT: 'var(--color-dark-background)',
		textfield: 'var(--color-dark-textfield-card)'
	},
```

You're the better judge here, I'll leave it upto you :)
### I'm not sure if we're using this name somewhere already. So it's best to do a little searching before the implementation.

**Review by chinmaykunkikar — CHANGES_REQUESTED** [review]

File: frontend/tailwind.config.js
Comment (SankalpHaritash21) [reply frontend/tailwind.config.js:10]: @chinmaykunkikar Hi,
I did some search and I found 
 extend: {
      colors: {
        dark: {
          background: {
            DEFAULT: 'var(--color-dark-background)',
            card: {
              textfield: 'var(--color-dark-textfield-card)',
            },
          },
        },
      },
    },
this way. I think this way will help to help other contributer in understanding.

and now I used it like " dark:bg-dark-background-card-textfield " and "dark:bg-dark-background "

And in above approach which  I use I  thought it may help other to understand.
can u review

Comment (chinmaykunkikar) [reply frontend/tailwind.config.js:10]: That looks good. You can go ahead with the suggestions now :)

Comment (SankalpHaritash21) [reply frontend/tailwind.config.js]: @chinmaykunkikar  I haven't updated PR as it is same

Comment (chinmaykunkikar) [reply frontend/tailwind.config.js]: Yeah, it wasn't needed. Thanks for the changes.

Comment (krishnaacharyaa) [reply frontend/tailwind.config.js]: @SankalpHaritash21, I actually agree with @chinmaykunkikar , background is redundant. Can we remove the background layer and use it like

```
extend: {
      colors: {
        dark: {
		DEFAULT: 'var(--color-dark-background)',
		textfield: 'var(--color-dark-textfield-card)'
	},
```
As suggested... This looks neat to me as we remove one more level of branching as background which is presently not required because `bg-dark-textfield` and `bg-dark` is more readable and better understandable than `dark:bg-dark-background-card-textfield` and `dark:bg-dark-background`

Kindly make the required changes :)

File: frontend/src/pages/details-page.tsx
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:18]: Kindly try using the tailwind-css recommended heights like h-48 which corresponds to 360 and 460 instead of arbitrary values

File: frontend/src/pages/details-page.tsx
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:25]: Please remove style and use equivalent classnames

File: frontend/src/pages/home-page.tsx
Comment (krishnaacharyaa) [inline frontend/src/pages/home-page.tsx:45]: Can we avoid using it multiple times by giving `dark:text-white` to the top parent div ?

File: frontend/tailwind.config.js
Comment (krishnaacharyaa) [inline frontend/tailwind.config.js:13]: kindly make the changes as commented

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Thank you for the dark theme support @SankalpHaritash21 it would really help our project. Kindly address the review comments, we are almost good to merge

File: frontend/tailwind.config.js
Comment (SankalpHaritash21) [reply frontend/tailwind.config.js]: sir can we keep "dark:bg-dark-background-textfield" background . Tryed but it creates a confusion. @krishnaacharyaa  its ur last decision.

Comment (krishnaacharyaa) [reply frontend/tailwind.config.js]: Please explain me why `dark:bg-dark-background-textfield` causes confusion and how is it better than @chinmaykunkikar's suggestion, will consider if convinced :) Because I feel the above is much clean and better to understand

Comment (SankalpHaritash21) [reply frontend/tailwind.config.js]: @krishnaacharyaa  I mean to say that is this background layer will help us to keep difference if we make any other changes in future then we can add a new different layer in easily in dark other wise we need to make changes in each file. 
As this background layer may seem to be irrelevant but it help to indicate background  color. 
Otherwise it create confusion as text-dark-textfield as we can use these color variable with both bg and text.

File: frontend/src/pages/details-page.tsx
Comment (krishnaacharyaa) [reply frontend/src/pages/details-page.tsx]: @SankalpHaritash21 please resolve

File: frontend/tailwind.config.js
Comment (krishnaacharyaa) [reply frontend/tailwind.config.js]: @SankalpHaritash21, Somehow i'm not convinced, I understand your concern, we'll change it when required in the future because for immediate future I dont see any need for the same, kindly make the change

File: frontend/src/pages/details-page.tsx
Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx]: className="h-80 w-full object-cover md:h-96" 
@krishnaacharyaa  as h-80 is equivalent to 320px and h-96 is equivalent to 384px above this we need to give custom height

Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx]: className="h-80 w-full object-cover md:h-96"
@krishnaacharyaa as h-80 is equivalent to 320px and h-96 is equivalent to 384px above this we need to give custom height

File: frontend/tailwind.config.js
Comment (SankalpHaritash21) [reply frontend/tailwind.config.js]: @krishnaacharyaa I have updated the code plz check

**Comment by krishnaacharyaa** [comment]
image

Deployment sees like this to me, doesn't match with the description of the PR :)

File: frontend/src/pages/home-page.tsx
Comment (krishnaacharyaa) [reply frontend/src/pages/home-page.tsx:45]: @SankalpHaritash21, please dont mark the conversation as resolved without actually resolving it. It's okay to ask us if you don't know or find it challenging we are here to help, by marking it falsely resolved you are just disrespecting the reviewer's time of reviewing the code. This is the second time I'm able to see this. If this follows I might have to decline the PR all together and assign it to some other potential candidate.

The analogy can be applied as it is just like I am rejecting your PR without any acknowledgement

Comment (SankalpHaritash21) [reply frontend/src/pages/home-page.tsx:45]: @krishnaacharyaa i thought I resolved it I will update it. And this will not happen again

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa  I have check in my system it is updates I thought I forget to upload components/ui folder.

File: frontend/src/pages/details-page.tsx
Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx]: @krishnaacharyaa is this issue resolved

File: frontend/tailwind.config.js
Comment (SankalpHaritash21) [reply frontend/tailwind.config.js]: @krishnaacharyaa same here is this issue also resolved

**Comment by SankalpHaritash21** [comment]
Hi, @krishnaacharyaa  plz review the code now it is working fine.

File: frontend/src/pages/home-page.tsx
Comment (SankalpHaritash21) [reply frontend/src/pages/home-page.tsx:45]: @krishnaacharyaa  I don't have any intention to disrespect the reviewer.

**Comment by krishnaacharyaa** [comment]
@SankalpHaritash21  Thanks for the update. Looks good to me, if you observe carefully the shimmer cards don't support the dark theme, Can you kindly raise a bugfix ticket, post that we will merge this :)

**Comment by SankalpHaritash21** [comment]
Ok

**Comment by chinmaykunkikar** [comment]
@krishnaacharyaa are we keeping the dark theme by default or are we giving users an option to toggle?

**Comment by krishnaacharyaa** [comment]
@chinmaykunkikar  This is the default dark theme based on the browser theme, for the toggle we have another issue

**Comment by chinmaykunkikar** [comment]
@krishnaacharyaa Got it, so we will be enabling toggling eventually.

@SankalpHaritash21 I was taking some personal time off. I'm reviewing this PR now.
Also, I request you to follow the commit message format mentioned in the contribution guidelines.
"Changes Updated", and "Changes needed" confuses others, while it doesn't tell me or Krishna anything about the commit.
It's possible that one might not know what to write as the message for the changes. In such cases, mention the PR id and mention some keywords from your last conversation about the requested changes.

**Comment by krishnaacharyaa** [comment]
@chinmaykunkikar, thanks for bringing that up. I was actually considering to add some constraints before making commits, didn't get any major leads, Do you know or tried? If such thing is possible. That would actually be really great for us because people unknowingly forget the format and there's no way out.

**Comment by chinmaykunkikar** [comment]
@krishnaacharyaa It's possible using husky and pre-commit hooks. But, let's not do that. Let's not make it a pain in the bum for the devs who have worked on something and want to commit their work.
We can explore some other solutions for what you're mentioning.

File: frontend/src/index.css
Comment (chinmaykunkikar) [inline frontend/src/index.css:8]: _Disclaimer: I haven't been following this conversation. So, apologies if things get repetitive_.

@krishnaacharyaa from where are we using this specific color? I don't see it anywhere in the official [tailwind theme colors](https://tailwindcss.com/docs/customizing-colors). The closest ones are either gray-900 or slate-900.

I suggest we stick to the official colors and change both of the colors to slate or gray schemes. Can you check it and confirm?

Comment (SankalpHaritash21) [reply frontend/src/index.css:8]: @chinmaykunkikar  I used this color from the figma file provided by the @krishnaacharyaa  
"https://www.figma.com/file/zqNcWGGKBo5Q2TwwVgR6G5/WanderLust--A-Travel-Blog-App?type=design&node-id=148-2&mode=design&t=ylnaldxFn5jMgyWI-0"

You can check.
First I tryed to used pre define color but these colors have some difference

Comment (SankalpHaritash21) [reply frontend/src/index.css:8]: @chinmaykunkikar  I used these specific color from the figma file provided by the @krishnaacharyaa 
 
https://www.figma.com/file/zqNcWGGKBo5Q2TwwVgR6G5/WanderLust--A-Travel-Blog-App?type=design&node-id=148-2&mode=design&t=ylnaldxFn5jMgyWI-0

you can check.

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa @chinmaykunkikar  I am trying to raise issue of Shimmer UI can you tell me what to add in 
Reproduction URL
Please enter your GitHub URL to provide a reproduction of the issue

I have no idea what to add

File: frontend/src/index.css
Comment (SankalpHaritash21) [reply frontend/src/index.css:8]: @chinmaykunkikar  can u resolve my issue too.

**Comment by SankalpHaritash21** [comment]
#64

File: frontend/src/index.css
Comment (chinmaykunkikar) [reply frontend/src/index.css:8]: @SankalpHaritash21 what issue would that be?

Comment (krishnaacharyaa) [reply frontend/src/index.css:8]: @chinmaykunkikar, actually @SankalpHaritash21 came up with closest tw colors (claimed) but it was too off,  So I gave him idea by asking him to refer dark theme referring to the tw official docs https://github.com/krishnaacharyaa/wanderlust/pull/43#issuecomment-1788456745 which looked cool to me, but i am still open to the official tw-colors as far it looks good.

**Comment by krishnaacharyaa** [comment]
> @krishnaacharyaa It's possible using husky and pre-commit hooks. But, let's not do that. Let's not make it a pain in the bum for the devs who have worked on something and want to commit their work. We can explore some other solutions for what you're mentioning.

@chinmaykunkikar, just a confirmation, how does it causes trouble to the devs? I meant when someone tries to commit which doesn't obey the predefined regex we give some friendly error message saying "Use this regex for the commit message". If we can get this done using the husky, then I think we are good. Happy to consider your recommendations :)

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa do we need any other changes or it look good

**Comment by krishnaacharyaa** [comment]
> Please enter your GitHub URL to provide a reproduction of the issue

@SankalpHaritash21 
Reproduction URL is the present deployment url,
Kindly give good reproduction steps, you need to say open url and then switch to the dark theme, then before the posts are loaded we see shimmer cards which are white.
Screenshot is not previewable, kindly update it

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa I will update it

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa what to add in screenshot

**Comment by chinmaykunkikar** [comment]
> @SankalpHaritash21
Reproduction URL is the present deployment url,
Kindly give good reproduction steps, you need to say open url and then switch to the dark theme, then before the posts are loaded we see shimmer cards which are white.
Screenshot is not previewable, kindly update it
> @krishnaacharyaa I will update it
> @krishnaacharyaa what to add in screenshot

Let's move the discussion of #64 to #64 please 😛

**Comment by chinmaykunkikar** [comment]
> @chinmaykunkikar, just a confirmation, how does it causes trouble to the devs? I meant when someone tries to commit which doesn't obey the predefined regex we give some friendly error message saying "Use this regex for the commit message". If we can get this done using the husky, then I think we are good. Happy to consider your recommendations :)

#65

**Comment by krishnaacharyaa** [comment]
@chinmaykunkikar, thank you for the quick fix, just a small observation, v1 looks good to me as the color looks subtle and doesn't pop off, but the blog post card's colour seems mismatching the featured and latest cards, can you kindly confirm on that, because sometimes images are different than actual colours, And I'm fine with your final call, this was my preference and observation ;)

Can we do it in seperate PR, helps us to keep it seperate commit, I'll squash the present PRs commit because of the commit messages...

File: frontend/src/index.css
Comment (SankalpHaritash21) [reply frontend/src/index.css:8]: Ok @krishnaacharyaa  then I will directly use these color 
@chinmaykunkikar can u tell me what's the background color,  card color and text field color

Comment (SankalpHaritash21) [reply frontend/src/index.css:8]: @krishnaacharyaa  which version should I use v1 Or v2

**Comment by SankalpHaritash21** [comment]
@chinmaykunkikar  plz let me first complete this task then you can do these changes plz

**Comment by krishnaacharyaa** [comment]
@SankalpHaritash21, I guess it @chinmaykunkikar already has fix for #64 , So i guess you can pick up some other story :)

File: frontend/src/index.css
Comment (krishnaacharyaa) [reply frontend/src/index.css:8]: @SankalpHaritash21 what version ? in the above context and he has clearly mentioned the colors with hex code and the tw-class names, kindly refer the same for the bg color , card color and text field color, This has solution to all of your problems https://github.com/krishnaacharyaa/wanderlust/pull/62#discussion_r1408422049

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa  I am saying that @chinmaykunkikar  can push changes after completing this issue

**Comment by chinmaykunkikar** [comment]
> Can we do it in seperate PR, helps us to keep it seperate commit, I'll squash the present PRs commit because of the commit messages...

Okay. I'll make those changes on a separate branch and push it to #64 once this is merged.

**Comment by chinmaykunkikar** [comment]
> P.S: I guess it's the shimmer which is giving the different colours 😂, my bad, V2's padding looks cool to me, feel free to take your call 😊

Correct 😆 
I'll see what looks the best here :)

**Comment by chinmaykunkikar** [comment]
@krishnaacharyaa @SankalpHaritash21

The only thing that's pending for this PR is the color variable names, right? What else?

I'll be taking up the skeleton ui task.

**Comment by krishnaacharyaa** [comment]
> @krishnaacharyaa @SankalpHaritash21
> 
> The only thing that's pending for this PR is the color variable names, right? What else?
> 
> I'll be taking up the skeleton ui task.

Yes @chinmaykunkikar, you are right

**Comment by krishnaacharyaa** [comment]
@SankalpHaritash21, please follow the above comment and finish this, lets merge this PR, it's been really long

**Comment by SankalpHaritash21** [comment]
https://github.com/krishnaacharyaa/wanderlust/pull/62#issuecomment-1832216987

@krishnaacharyaa  I will try to complete it by today.

**Comment by SankalpHaritash21** [comment]
@chinmaykunkikar @krishnaacharyaa  I used this approach 

/** @type {import('tailwindcss').Config} */

export default {
  content: ['./index.html', './src/**/*.{js,ts,jsx,tsx}'],
  darkMode: 'media',
  theme: {
    extend: {
      colors: {
        dark: {
          DEFAULT: '#0f172a',//bg-slate-900
          textfield: '#1e293b',//bg-slate-800
        },
      },
    },
  },
  plugins: [],
};

and removed code from index.css.
Can u plz

**Comment by SankalpHaritash21** [comment]
@chinmaykunkikar @krishnaacharyaa  check then I will push the code.

**Comment by krishnaacharyaa** [comment]
Looks good to me, but I see the tw-classes are commented out, if the image is of the commented classes then fine

**Comment by SankalpHaritash21** [comment]
https://github.com/krishnaacharyaa/wanderlust/pull/62#issuecomment-1833820736

@krishnaacharyaa  I tryed to use tw-classes names but it is breaking the code and I also read tw docs but I found this way is effective that's why I use hex code of slate-900 and slate-800.

This is @krishnaacharyaa @chinmaykunkikar  your decision. If both of you seems this method to useful let me know I will push the code.

**Comment by SankalpHaritash21** [comment]
const colors = require('tailwindcss/colors')

module.exports = {
  theme: {
    colors: {
dark: {
DEFAULT: colors.slate,
textfield: color.slate,
}
    },
  },
}


@krishnaacharyaa  I tryed this approach also. but it I am not able to complete
If you can solve this issue you can. I failed to complete this issue.

**Comment by SankalpHaritash21** [comment]
@krishnaacharyaa  can i take home page issue if u like

**Comment by chinmaykunkikar** [comment]
@SankalpHaritash21 don't worry. You came a long way. Kudos! :)
@krishnaacharyaa please merge this commit, I'll do these small changes in the skeleton PR.

**Review by krishnaacharyaa — APPROVED** [review]
Merging it now :)

**Pull request merged**
